controllers/configurationpolicy_controller.go

```diff
@@ -932,6 +932,23 @@ func sortRelatedObjectsAndUpdate(
 		}
 	}
 
+	// add RelatedObject into newRelated which is in oldRelated but not in newRelated
+	for _, oldEntry := range oldRelated {
+		ifAppend := true
```

Comment:
Nit: rename to `shouldAppend`. Or possibly reverse it and call it `found`, so that the end is "if not found, append".